### PR TITLE
Introduce error code for when a PaymentPoint cannot afford a refund

### PR DESCRIPTION
**Pull request by Dbk-JungleBoy**
Added new errorcode to Refund. When a PaymentPoint does not have the sufficient funds to create a refund, we will return 409 statuscode instead of 500. 

Extended Refund flow description

### CODE REVIEW

File: docs/payment_flows.md [hunk 1]
```diff
@@ -61,7 +61,8 @@ It is required of the client to implement a periodically scheduled job of runnin
 Once a payment is captured, the payment amount is immediately charged from the customer. Therefore, the payment can no
 longer be cancelled. Instead, the payment amount can be transferred back to the customer, by performing *refunds*. 
 Each captured payment can be refunded multiple times with the restriction that the sum of the refunds cannot exceed
-the captured payment amount. A payment can be refunded up to 30 days after the payment was captured. After 30 days a refund is no longer possible with MobilePay.
+the captured payment amount. A payment can be refunded up to 90 days after the payment was captured. After 90 days a refund is no longer possible with MobilePay.
+A refund can only be made if there is sufficient funds on the payment point.
```
Comment (Dbk-JungleBoy) [inline docs/payment_flows.md]: Better description about when refunds is not possible :)

File: docs/endpoint_errors.md [hunk 2]
```diff
@@ -168,7 +168,7 @@ This page contains information regarding all the non-successful status-codes and
...
 | 403 | 1401 | Cannot refund payments created by a different integrator |
-| 409 | 1001 <br> 1306`` `` <br> <br> 1354 <br> 1365 <br> 1366 <br> 1367 | Payment not found <br> ``x-mobilepay-idempotency-key`` header has to be unique per request unless the request is a retry of a previous request <br> Refund of payment not possible when payment is not captured <br> Refund ``CurrencyCode`` is different than payment ``CurrencyCode`` <br> Payment is too old <br> Refund ``Amount`` is too high |
+| 409 | 1001 <br> 1306`` `` <br> <br> 1354 <br> 1365 <br> 1366 <br> 1367 <br> 1368 | Payment not found <br> ``x-mobilepay-idempotency-key`` header has to be unique per request unless the request is a retry of a previous request <br> Refund of payment not possible when payment is not captured <br> Refund ``CurrencyCode`` is different than payment ``CurrencyCode`` <br> Payment is too old <br> Refund ``Amount`` is too high <br> Refund not possible due to insufficient funds on payment point |
```
Comment (Dbk-JungleBoy) [inline docs/endpoint_errors.md:171]: Better description needed for when one can't create a refund due to insufficient funds

File: docs/payment_flows.md [hunk 3]
```diff
@@ -61,7 +61,8 @@ It is required of the client to implement a periodically scheduled job of runnin
 Once a payment is captured, the payment amount is immediately charged from the customer. Therefore, the payment can no
 longer be cancelled. Instead, the payment amount can be transferred back to the customer, by performing *refunds*. 
 Each captured payment can be refunded multiple times with the restriction that the sum of the refunds cannot exceed
-the captured payment amount. A payment can be refunded up to 30 days after the payment was captured. After 30 days a refund is no longer possible with MobilePay.
+the captured payment amount. A payment can be refunded up to 90 days after the payment was captured. After 90 days a refund is no longer possible with MobilePay.
+A refund can only be made if there is sufficient funds on the payment point (wallet).
```
Comment (Dbk-JungleBoy) [inline docs/payment_flows.md:65]: Better description needed for when one can't create a refund due to insufficient funds

File: docs/payment_flows.md [hunk 4]
```diff
@@ -61,7 +61,9 @@ It is required of the client to implement a periodically scheduled job of runnin
 Once a payment is captured, the payment amount is immediately charged from the customer. Therefore, the payment can no
 longer be cancelled. Instead, the payment amount can be transferred back to the customer, by performing *refunds*. 
 Each captured payment can be refunded multiple times with the restriction that the sum of the refunds cannot exceed
-the captured payment amount. A payment can be refunded up to 30 days after the payment was captured. After 30 days a refund is no longer possible with MobilePay.
+the captured payment amount. A payment can be refunded up to 90 days after the payment was captured. After 90 days a refund is no longer possible with MobilePay.
+
+A refund can be executed if the store payment balance contains enough money to cover the refund. If the balance doesn’t cover the amount of the refund, the refund will fail.
```
Comment (ABEK-MP) [inline docs/payment_flows.md:66]: Maybe clarify *how* the refund will fail in this case? ie. "the refund will fail [with error code XXXX]"

**Pull request merged**